### PR TITLE
Rewards UI cleanup

### PR DESCRIPTION
**Pull request by christianbaroni**
## What changed (plus any additional context for devs)
- Memoizes the cards on the Rewards screen
- Cleans up the next drop time, now shows e.g. "Today 4:20 PM" instead of "Tuesday 4:20pm" if the drop is happening at 4:20 PM on the current day
- Trims trailing zeros from "Rainbow users have earned" amount
- Fixes the rainbow text gradient width in the "My Points" section in the case where the device uses a non-standard system font size
- Cleans up the pending transactions spinner in the tab bar
- Fixes a bug causing the `SmoothPager` wrapper views to partly block the tap region behind the sheet in the case where the `SmoothPager` page heights differ
- Removes default exports in main points components

### CODE REVIEW

File: src/screens/points/content/PointsContent.tsx [hunk 1]
```diff
@@ -476,15 +477,17 @@ const NextDistributionCountdown = ({ nextDistribution }: { nextDistribution: Dat
 
   return (
     <Text align="center" color="labelSecondary" size="17pt" weight="heavy">
-      {`${dayStr} ${hourStr} ${minuteStr}`}
+      {`${dayStr} ${hourStr} ${minuteStr}`.trim()}
```
Comment (brunobar79) [inline src/screens/points/content/PointsContent.tsx:480]: Where is the whitespace coming from?

**Review by brunobar79 — APPROVED** [review]
LGTM!

File: src/screens/points/content/PointsContent.tsx (hunk 1 above)
Comment (christianbaroni) [reply src/screens/points/content/PointsContent.tsx:480]: The `dayStr` is empty within 24h of the drop, and `hourStr` is empty within 1h of the drop, so the whitespace between d/h/m end up being included without `.trim()`

Comment (brunobar79) [reply src/screens/points/content/PointsContent.tsx:480]: makes sense!

**Pull request merged**